consensus/core/src/network/anemo_network.rs

```diff
@@ -0,0 +1,335 @@
+        self.service
+            .handle_send_block(*index, block)
+            .await
+            .map_err(|e| {
+                anemo::rpc::Status::new_with_message(
+                    anemo::types::response::StatusCode::BadRequest,
+                    format!("{e}"),
+                )
+            })?;
+        Ok(Response::new(SendBlockResponse {}))
+    }
+
+    async fn fetch_blocks(
+        &self,
+        request: anemo::Request<FetchBlocksRequest>,
+    ) -> Result<anemo::Response<FetchBlocksResponse>, anemo::rpc::Status> {
+        let Some(peer_id) = request.peer_id() else {
+            return Err(anemo::rpc::Status::new_with_message(
+                anemo::types::response::StatusCode::BadRequest,
+                "peer_id not found",
+            ));
+        };
+        let index = self.peer_map.get(peer_id).ok_or_else(|| {
+            anemo::rpc::Status::new_with_message(
+                anemo::types::response::StatusCode::BadRequest,
+                "peer not found",
+            )
+        })?;
+        let block_refs = request.into_body().block_refs;
+        let blocks = self
+            .service
+            .handle_fetch_blocks(*index, block_refs)
+            .await
+            .map_err(|e| {
+                anemo::rpc::Status::new_with_message(
+                    anemo::types::response::StatusCode::BadRequest,
+                    format!("{e}"),
+                )
+            })?;
+        Ok(Response::new(FetchBlocksResponse { blocks }))
+    }
+}
+
+pub(crate) struct AnemoManager {
+    context: Arc<Context>,
+    client: Arc<AnemoClient>,
+}
+
+#[allow(unused)]
+impl AnemoManager {
+    pub(crate) fn new(context: Arc<Context>) -> Self {
+        Self {
+            context: context.clone(),
+            client: Arc::new(AnemoClient::new(context)),
+        }
+    }
+}
+
+impl NetworkManager<AnemoClient, AnemoService> for AnemoManager {
+    fn client(&self) -> Arc<AnemoClient> {
+        self.client.clone()
+    }
+
+    fn install_service(&self, network_signer: NetworkKeyPair, service: Box<AnemoService>) {
+        let server = ConsensusRpcServer::new(ServiceProxy::new(self.context.clone(), service));
+        let authority = self.context.committee.authority(self.context.own_index);
+        let address = authority
+            .address
+            .clone()
+            .replace(0, |_protocol| Some(Protocol::Ip4(Ipv4Addr::UNSPECIFIED)))
+            .unwrap();
+        let all_peer_ids = self
+            .context
+            .committee
+            .authorities()
+            .map(|(_i, authority)| anemo::PeerId(authority.network_key.0.to_bytes()));
+        let routes = anemo::Router::new()
+            .route_layer(RequireAuthorizationLayer::new(AllowedPeers::new(
+                all_peer_ids,
+            )))
+            .add_rpc_service(server);
+        // TODO: add layers.
+        let service = tower::ServiceBuilder::new().service(routes);
+
+        // TODO: add peer filter.
```

Comment:
Let's make sure that we also add the epoch filters etc

Reply:
oh sure , by no means is byzantine robust, but it's a quick way to ensure that nodes across different epochs (ex during the epoch change period) are not communicating and avoid issues from avoid spending resources (ex due to synchronizaton)

Reply:
Discussed offline and we can definitely add it.